lib/web/apiserver.go

```diff
@@ -910,6 +920,69 @@ func getAuthSettings(ctx context.Context, authClient auth.ClientI) (webclient.Au
 	return as, nil
 }
 
+// traces forwards spans from the web ui to the upstream collector configured for the proxy. If tracing is
+// disabled then the forwarding is a noop.
+func (h *Handler) traces(w http.ResponseWriter, r *http.Request, _ httprouter.Params, _ *SessionContext) (interface{}, error) {
+	var data tracepb.TracesData
+	b, err := io.ReadAll(r.Body)
```

Comment:
Should we use a limited reader here to prevent allocating massive amounts of memory?